File: holoken-core/src/main/java/com/srlee/dlx/DLXColumn.java
package com.srlee.dlx;

class DLXColumn extends LL2DNode {
	private int size;        // Number of items in column
	
	DLXColumn() {
		size = 0;
		up = this;
		down = this;
	}
	
	int getSize() {
		return size;
	}
	
	void decrementSize() {
		size--;
	}
	
	void incrementSize() {
		size++;
	}
}

File: gauguin-core/src/main/kotlin/org/piepmeyer/gauguin/creation/dlx/DLXColumn.kt
package org.piepmeyer.gauguin.creation.dlx

internal class DLXColumn : LL2DNode() {
    // Number of items in column
    var size = 0
            private set

    init {
        up = this
        down = this
    }

    fun decrementSize() {
        size--
    }

    fun incrementSize() {
        size++
    }
}
